### DISCUSSION TITLE
customize friendly_name

### DISCUSSION THREAD

**Discussion by spali** [zachowj/hass-node-red#54] [Q&A] [answered]
Could also be in a question for the node-red package. But maybe the Integration API is the limit here.

I had success with creating dynamic entities with `nodered/discovery` with the API node from the node-red.
Same for updating with `nodered/entity` to set state and attributes.

But I would like to set the `name` (entity_id) and `friendly_name` independent of each other.
Reason is I try to keep all entities in English, but would like to customize the `friendly_name` in German.
I know, I could set this in HA itself and overwrite the `friendly_name`, but this would break with my dynamic creation of the entities and I like to have the creation reproducible.

Is it possible to specifically set the firendly_name in  `nodered/discovery` or `nodered/entity`  that is different from the `name` ?

I tried already with updating `firendly_name` as attribute and also changing config after creation, but no luck. Is this somehow supported or would this be a good feature request? ;)

**Comment by zachowj** [comment, marked as answer]
The first time you register an entity using `nodered/discovery` it uses the name field to create an `entity_id` based on the name field or a combination of the `server_id` and `node_id`. Any updates sent to `nodered/discovery` using the same `server_id` and `node_id` will only change the name field and not the `entity_id`.

You could do back-to-back API calls to `nodered/discovery` with the second one containing the German name in the name property. Not the best solution but should work for what you're trying to do.

**Comment by spali** [comment]
thank you so much, it worked.
Didn't got the idea to reuse the `nodered/discovery` call a second time... I already had a back-to-back API call with `nodered/entity`, but that didn't work obviously. But second call to `nodered/discovery` worked as expected.
